api-reference/10 UI Widgets/dxGantt/1 Configuration/tasks/endExpr.md

```diff
@@ -8,3 +8,141 @@ default: 'end'
 Specifies the data field that provides tasks' end dates.
 
 ---
+---
+##### jQuery
+
+    <!-- tab: index.js -->
+    $(function() {
+        $("#gantt").dxGantt({
+            tasks: {
+                dataSource: tasks,
+                startExpr: "startDate",
+                endExpr: "endDate"
+            },
+            //...
+        });
+    });
+    <!-- tab: data.js -->
+    var tasks = [{
+        'id': 1,
+        'parentId': 0,
+        'title': 'Software Development',
+        'startDate': new Date('2019-02-21T05:00:00.000Z'),
+        'endDate': new Date('2019-07-04T12:00:00.000Z'),
+        'progress': 31
+    },
+    // ...
+    ];
+
+##### Angular
+
+    <!-- tab: app.component.ts -->
+    import { DxGanttModule } from 'devextreme-angular';
+    import { Service, Task, Dependency, Resource, ResourceAssignment } from './app.service';
+    // ...
+    export class AppComponent {
+        tasks: Task[];
+        // ...
+    }
+    <!-- tab: app.component.html -->
+    <dx-gantt ... 
+        <dxo-tasks [dataSource]="tasks" [endExpr]="endDate" [startExpr]="startDate" ></dxo-tasks>
+		// ...
```

Comment:
Comments in the markup use the following syntax:
```
<!-- comment -->
```
In this case, the line should be

```
<!-- ... -->
```